### PR TITLE
Feature: Custom admin login image

### PR DESCRIPTION
**Pull request by bahiirwa**
WIP on login image. #499 

Changes to come to description when full submission is done.

### CODE REVIEW

File: src/wp-includes/option.php [hunk 1]
```diff
@@ -1847,15 +1847,26 @@ function register_initial_settings() {
...
 		'type'         => 'boolean',
 		'default'      => false,
 		'description'  => __( 'Use the site logo as the login image' ),
 	) );
 
+	register_setting( 'general', 'login_custom_logo', array(
+		'show_in_rest' => array(
+			'name' => 'login_custom_logo',
+			'schema'  => array(
+				'format' => 'uri',
```
Comment (nylen) [inline src/wp-includes/option.php]: This should probably be an integer (attachment ID) instead of a URI?

Comment (nylen) [reply src/wp-includes/option.php]: Thought about it some more and yes, this needs to be updated to store an attachment ID. If an image URI is stored, then this feature will break when a site is moved to a different domain for example.

**Comment by bahiirwa** [comment]
@nylen 
1. Is this the best place to have our JavaScript? Does the common.js file need a dependency of some kind?
1. Am stuck with how to bring in wp.media. See screenshot.

Screenshot 2020-02-27 at 22 39 46

**Comment by nylen** [comment]
1. I would probably start a new JavaScript file instead. `src/wp-admin/js/options-general.js` ?
2. It looks like you need to call [`wp_enqueue_media()`](https://codex.wordpress.org/Function_Reference/wp_enqueue_media) in order to enqueue the media scripts. This appears to be a pretty good guide for how to get everything working, the only difference would be that it tells how to do this in a plugin instead of in the core code: https://wordpress.stackexchange.com/questions/235406/how-do-i-select-an-image-from-media-library-in-my-plugin

**Comment by bahiirwa** [comment]
>     1. I would probably start a new JavaScript file instead. `src/wp-admin/js/options-general.js` ?

I was debating this.

>     2. It looks like you need to call [`wp_enqueue_media()`](https://codex.wordpress.org/Function_Reference/wp_enqueue_media)

I am now debating whether we need to add this in `src/wp-includes/script-loader.php` as a standalone file. What if the need grows to past just a media uploader?

I have since taken the approach in the commits. But the above question still lingers. A future change could probably address it.

**Comment by bahiirwa** [comment]
I am stuck here. 

1. Using the returned data in the hidden field and saving it in the DB

**Comment by nylen** [comment]
I've pulled this in to the `feature/login-image` branch on the main repository and I'll continue developing from there.

**Comment by bahiirwa** [comment]
Thanks for taking this up.

**Pull request merged**